sdk/core/azure-core-test/src/main/java/com/azure/core/test/annotation/DoNotRecord.java

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.test.annotation;
+
+import com.azure.core.test.TestMode;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Annotation given to some tests to indicate that network calls made during the test shouldn't be recorded.
+ *
+ * <p>
+ * Pass {@code true} for {@link #skipInPlayback() skipInPlayback} to indicate that the test shouldn't run when tests are
+ * ran in {@link TestMode#PLAYBACK}. A common case for setting this to {@code true} is when the test has either
+ * sensitive content that cannot be redacted or calls into code that cannot be mocked.
+ */
+@Retention(RUNTIME)
+@Target({METHOD})
+public @interface DoNotRecord {
```

Comment:
Just for my understanding, why would any test run only in live mode but shouldn't be recorded?

Reply:
Tests that contain information that cannot be recorded into a playback JSON such as random UUIDs generated by methods called.